### DISCUSSION TITLE
Is there a way to "maybe" select a possibly-nonexisting column?

### DISCUSSION THREAD

**Discussion by erezinman** [duckdb/duckdb#14075] [Q&A]
I wish to read a column only if it exists. However, none of the variable-columns methods I tried work. 
Examples:
```sql
SELECT col1, COLUMNS(c -> c='col2')
FROM ...
```
results in 
> Binder Error: Star expression "COLUMNS(list_filter([...], (c -> (c = 'col2'))))" resulted in an empty set of columns

while, 
```sql
SELECT col1, COLUMNS('col2')
FROM ...
```
results in 
> Binder Error: No matching columns found that match regex "col2"

To be more specific, I select from a parquet file.

Any ideas?

**Comment by Tishj** [comment]
`select COLUMNS([x for x in (*) if x in ['a', 'b', 'c'])`

where you replace `'a', 'b', 'c'` with the list of columns you want to select when the source contains them.
Note that this will likely fail when none of those columns are available

**Reply by erezinman** [reply]
Your comment in the end is exactly the issue I'm facing. :-|

**Reply by Tishj** [reply]
`select COLUMNS([x for x in (*) if x in ['col1', 'col2'])` sounds like it should fix your problem? If `col1` is always present

**Reply by erezinman** [reply]
This might work, but when I want something more complex that also includes aliasing the column, this wouldn't work.

**Comment by erezinman** [comment]
I think I got this
```
CREATE TABLE flights AS
    FROM 'https://duckdb.org/data/flights.csv';


WITH temp_table as (
    SELECT NULL as temp_col_col_col
)
SELECT first(coalesce(*COLUMNS(c -> c in ['FlightDate', 'temp_col_col_col']))) as 'FlightDate'
FROM flights 
cross join temp_table;

WITH temp_table as (
    SELECT NULL as temp_col_col_col
)
SELECT first(coalesce(*COLUMNS(c -> c in ['a', 'temp_col_col_col']))) as 'a'
FROM flights 
cross join temp_table;
```

**Reply by erezinman** [reply]
Note that this is still not perfect, because it allows me to select that column if exists or null, but it doesn't allow me to not select it.
